### PR TITLE
'special_nochildren' mailboxes should have \Noinferiors attribute

### PR DESCRIPTION
**Pull request by ksmurchison**
If we don't allow children to be created, then the mailbox should have \Noinferiors

### CODE REVIEW

File: imap/imapd.c [hunk 1]
```diff
@@ -13325,6 +13326,34 @@ static void add_intermediates(const char *extname, struct list_rock *lrock)
     strarray_fini(&inter);
 }
 
+static int is_noinferiors(struct findall_data *data, struct list_rock *rock)
+{
+    int r = 0;
+
+    if (data->mb_category == MBNAME_ALTINBOX)
```
Comment (rsto) [inline imap/imapd.c:13333]: nit: I suggest to also add block-braces here, or handle the else condition first, followed by an else with the single statement. What we really should to do is agree on what to do in that case in the style guide. Note that the style guide already forbids an empty line between if and else blocks.

File: imap/imapd.c [hunk 2]
```diff
@@ -13325,6 +13326,34 @@ static void add_intermediates(const char *extname, struct list_rock *lrock)
...
+        r = 1;
+
+    else if (strarray_size(rock->special_nochildren)) {
+        struct buf attrib = BUF_INITIALIZER;
+
+        annotatemore_lookup(mbname_intname(data->mbname), "/specialuse",
+                            mbname_userid(data->mbname), &attrib);
+
+        if (buf_len(&attrib)) {
+            strarray_t *uses = strarray_split(buf_cstring(&attrib), NULL, 0);
```
Comment (rsto) [inline imap/imapd.c:13343]: nit: in this block I find the empty lines to make it harder to read. Also in the block above. They are not logically separate statements, so let's keep them together.

**Review by rsto — CHANGES_REQUESTED** [review]
The logic looks good to me, but I have remarks on the code style.

File: imap/imapd.c (hunk 1 above)
Comment (ksmurchison) [reply imap/imapd.c:13336]: Fixed

File: imap/imapd.c (hunk 2 above)
Comment (ksmurchison) [reply imap/imapd.c:13345]: Fixed

**Review by rsto — APPROVED** [review]
LGTM

**Pull request merged**